ui/src/app/pages/admin/admin-user-common.tsx

```diff
@@ -221,6 +222,41 @@ export const displayTierBadgeByRequiredModule = (
   );
 };
 
+export const getEraNote = (profile: Profile): string => {
+  // Boolean array that indicates whether eRA is required for Registered Tier and Controlled Tier
```

Comment:
thanks!  I understand what's going on here now.

Unfortunately we're making some assumptions here that I'd prefer not to rely on:
* AccessTierShortNames and AccessTierDisplayNames are in the same order
* the order of these enums is the same as how we want to display them in the ERA Note

What I'd like to see here is a direct expression in code of what we're trying to do here.  I think we can achieve this in a clear, functional way.

So let's start with: what do we want?

* Display [RT needed] and [CT needed] in that order, or different text if neither.

Rendering an array of AccessTierDisplayName looks like a great choice to me, so I am happy with everything below the definition of `requiredTierDisplayNames`.  But I think we can improve on how we create this array.  So what do we want here?

* An array of 0, 1, or 2 of ['Registered Tier', 'Controlled Tier'] depending on which is required here, and in that order if both.

We have the functions we need to build this: which tiers are needed, and convert Short Name to Display Name.  So conceptually it's:
```
tierShortNamesInDesiredOrder
  .filter(isEraRequiredForTier)
  .map(displayNameForTier)
```

Reply:
A cleaner function, with Mohs's help.